src/amo/components/SearchForm/index.js

```diff
@@ -114,20 +114,32 @@ export class SearchFormBase extends React.Component {
     this.setState({ searchValue: event.target.value });
   }
 
-  handleSuggestionsFetchRequested = this.props.debounce(({ value, reason }) => {
+  dispatchAutocompleteStart = this.props.debounce(({ filters }) => {
+    const { dispatch, errorHandler } = this.props;
+
+    dispatch(autocompleteStart({
+      errorHandlerId: errorHandler.id,
+      filters,
+    }));
+    this.setState({ oldAutocompleteFilters: filters });
+  }, 200, { leading: true })
```

Comment:
I know you're relying on defaults here but since you are already specifying an option, it looks like the `debounce` is configured for leading edge only. I would suggest adding them both: `{ leading: true, trailing: true }`.

Reply:
I inverted `leading` and `trailing` it seems.